src/puya/awst/nodes.py

```diff
@@ -1499,6 +1505,43 @@ def accept(self, visitor: ModuleStatementVisitor[T]) -> T:
         return visitor.visit_logic_signature(self)
 
 
+@attrs.frozen
+class CompiledContract(Expression):
+    contract: ContractReference
+    allocation_overrides: Mapping[CompiledReferenceField, Expression] = attrs.field(
```

Comment:
could we use TxnField here instead for consistency? we don't need to validate the fields used either  I don't think - that can be done in the builder / frontends